### DISCUSSION TITLE
Query builder: date filter

### DISCUSSION THREAD

**Discussion by urusha** [grafana/clickhouse-datasource#579] [Ideas]
Please, add support for date column filter like `WITHIN DASHBOARD DATE RANGE` in the query builder. We have `$__dateFilter(columnName)` which is very useful for speeding up queries with date-partitioned tables, but it is not available in the builder.

**Comment by SpencerTorres** [comment]
Not sure which version you're using, but this is already a feature. It should be available as long as you have a time-like column selected for the filter:

![image](https://github.com/grafana/clickhouse-datasource/assets/28887171/a5f44000-ba51-48ea-b677-a9a456487c9d)

This will append `WHERE ( timestamp >= $__fromTime AND timestamp <= $__toTime )` to the query, where `$__*Time` is a `DateTime` type. Do you mean this should use a `Date` type? In that case we would need to add a macro that returns the dashboard time as a `Date` and call it `$__fromDate` and `$__toDate`. (feel free open an issue if you think these should be added)

You could also just write your own cast in manually:
![image](https://github.com/grafana/clickhouse-datasource/assets/28887171/14171563-cd39-4c49-9e2b-398401ffa53c)

**Comment by urusha** [comment]
> Not sure which version you're using, but this is already a feature. It should be available as long as you have a time-like column selected for the filter:

version 10.4.1
> 
> ![image](https://private-user-images.githubusercontent.com/28887171/317774331-a5f44000-ba51-48ea-b677-a9a456487c9d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NDM0OTEsIm5iZiI6MTcxMTY0MzE5MSwicGF0aCI6Ii8yODg4NzE3MS8zMTc3NzQzMzEtYTVmNDQwMDAtYmE1MS00OGVhLWI2NzctYTlhNDU2NDg3YzlkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDE2MjYzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZhODBmODQ5NDYyNWNmM2I2ZDkxYjRmMmQ2YmMyMGQyNzdkNTFkMTE2MzIzOWI1MzMxZWFiYWUxZTgwOTkyMzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hDnPF9wxy0UzNA004LAbEzUoZrfyFhDvq-q15Mdtc1k)
> 
> This will append `WHERE ( timestamp >= $__fromTime AND timestamp <= $__toTime )` to the query, where `$__*Time` is a `DateTime` type. Do you mean this should use a `Date` type? In that case we would need to add a macro that returns the dashboard time as a `Date` and call it `$__fromDate` and `$__toDate`. (feel free open an issue if you think these should be added)
> 
Yes, I'm talking about `Date` type - it filters out unneeded partitions.

> You could also just write your own cast in manually: ![image](https://private-user-images.githubusercontent.com/28887171/317776318-14171563-cd39-4c49-9e2b-398401ffa53c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NDM0OTEsIm5iZiI6MTcxMTY0MzE5MSwicGF0aCI6Ii8yODg4NzE3MS8zMTc3NzYzMTgtMTQxNzE1NjMtY2QzOS00YzQ5LTllMmItMzk4NDAxZmZhNTNjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDE2MjYzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI3OWU3ZGQ2OWM1M2NjZGNhNDhiMDE1ZmQxNjY2NmEwMGMxZjUwNzViNTYzYjE5ZWMwNTE4ZGEyOWNjNmQ0ZjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZzeI3P3qu2JY9sEXjI2Xvk2kUtBy1MFmrrBtjeYfhp4)

I'm not able to save such input - it disappears, getting `No options found`

**Reply by SpencerTorres** [reply]
What version of the plugin are you using?

As long as you select an operator (such as `>=` or `<=`) you should be able to write in your own SQL for the filter value. At least as of plugin `v4.x`

**Reply by urusha** [reply]
Thanks, that works after plugin update.
